pkg/util/ecs/common.go

```diff
@@ -23,6 +25,10 @@ import (
 // their metrics) for the current task by collecting that information from the
 // ECS metadata v2 API.
 func ListContainersInCurrentTask() ([]*containers.Container, error) {
+	if !config.IsCloudProviderEnabled(CloudProviderName) {
```

Comment:
This call is in the hot path to retrieve containers, I think we should instead disable the overall collector 
https://github.com/DataDog/datadog-agent/blob/master/pkg/util/containers/collectors/detector.go